server/member/member.go

```diff
@@ -77,7 +77,7 @@ func (m *Member) getLeader(ctx context.Context) (*getLeaderResp, error) {
 		return nil, ErrInvalidLeaderValue.WithCause(err)
 	}
 
-	return &getLeaderResp{Leader: leader, Revision: leaderKv.ModRevision, IsLocal: leader.GetId() == m.ID}, nil
+	return &getLeaderResp{Leader: leader, Revision: leaderKv.ModRevision, IsLocal: leader.GetEndpoint() == m.Endpoint}, nil
```

Comment:
- Add a method `isLocal` to encapsulate this logic.
- Why change id check to endpoint?

Reply:
The original implementation uses etcd id to determine whether it is local node. Now when using an external etcd cluster, you need to judge by the registered node address.